### PR TITLE
Add automatic builds in travis-ci

### PR DESCRIPTION
**Pull request by boxdot**
Also:
* Include Dockerfile of the image used for building and testing.
* Search for gtest package also as module.

### CODE REVIEW

File: flatdata-cpp/test/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-find_package(GTest CONFIG QUIET)
```
Comment (imagovrn) [inline flatdata-cpp/test/CMakeLists.txt]: Why this change?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+sudo: required
+language: cpp
+
+services:
+  - docker
+
+before_install:
+  - docker pull boxdot/flatdata-dev
```
Comment (imagovrn) [inline .travis.yml]: Would it be possible to use Dockerfile from source tree? Otherwise, who is able to and how to update the image?

File: flatdata-cpp/test/CMakeLists.txt (hunk 1 above)
Comment (boxdot) [reply flatdata-cpp/test/CMakeLists.txt]: To be able to find the package, if it was installed system-wide as a module.

File: .travis.yml (hunk 2 above)
Comment (boxdot) [reply .travis.yml]: Do you mean to build the image for each test run? This would take too much time since pandas is compiled inside the docker natively. And, most likely, we want to keep the testing environment stable.

If you mean, to build the image on https://hub.docker.com/ from the Dockerfile automatically. Yes, this is possible, however, I don't have permission to add this repository to my dockerhub account.

Comment (imagovrn) [reply .travis.yml]: @boxdot With respect to building image each test run, sounds reasonable. For second question: sorry, I meant, if you suddenly go windsurfing for two weeks, we wouldn't be able to update image at `boxplot/flatdata-dev`, right? Would be cool if there would be `heremaps/flatdata-dev` or otherwise we might want to consider creating `flatdata/dev`. Or am I going overboard?

File: ci/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# CI image for building with gcc and clang
+FROM alpine:latest
+
+RUN apk --no-cache add clang cmake make binutils musl-dev g++ boost-dev gtest gtest-dev gfortran python3-dev && \
```
Comment (imagovrn) [inline ci/Dockerfile]: Do we really need `musl-dev` and `gfortran`? O_O

Comment (boxdot) [reply ci/Dockerfile]: `musl-dev`: alpine comes with musl libc, we need it since it also contains the runtime object files, needed for creation of elf files.

`gfortran` is needed for compiling pandas (or deps of pandas). Probably, we should consider making pandas optional for building and testing, then we could remove `gfortran`. Also pip3 installations would be obsolete, since the other python packages are part of the alpine distro.

File: .travis.yml (hunk 2 above)
Comment (boxdot) [reply .travis.yml]: > With respect to building image each test run, sounds reasonable.

Not sure about that. Sounds wasteful, and also dangerous since a failure during the build phase of the image does not necessarily mean that something is wrong with flatdata. Further, an external update of a dependency can break the build. Partially because of that, `Gemfile.lock` (ruby) and `package-lock.json` (npm) were introduced.

> Would be cool if there would be heremaps/flatdata-dev or otherwise we might want to consider creating flatdata/dev. Or am I going overboard?

Not really, I totally agree with your concerns. The best case scenario is when `flatdata-dev` image is built automatically on dockerhub based on the Dockerfile from the source code. A new build should be triggered, when `Dockerfile` is updated in source.

I propose, for now, to go with my intermediate solution. And discuss with `heremaps` a possibility of automatic `heremaps/flatdata-dev` image.

Comment (imagovrn) [reply .travis.yml]: > Not sure about that.
I meant your comment sounds reasonable :)
> I propose, for now, to go with my intermediate solution.
As long as this doesn't become permanent solution :) Could you create an issue for us to remember about it?

File: ci/Dockerfile (hunk 3 above)
Comment (imagovrn) [reply ci/Dockerfile]: I see. Thanks! +2

**Comment by boxdot** [comment]
As follow-up: #11
